tpm/constants.go

```diff
@@ -123,6 +127,10 @@ const (
 	algXOR
 )
 
+func (a Algorithm) String() string {
+	return [...]string{"", "RSA", "", "", "SHA", "HMAC", "AES128", "MGF1", "AES192", "AES256", "XOR"}[a]
```

Comment:
This makes me a bit nervous about out-of-bounds panics when a new const is added.
How about using a map[Algorithm]string and handling unknown algs explicitly?

Reply:
All algs specified are already implemented as const here.

See: TPM-Main-Part-2-TPM-Structures_v1.2_rev116_01032011, Page 18.
 
So noone should get the idea of adding new algs as const, at least as long as he sticks to the specification.

Reply:
There's still possibility for user error:
```
id := someFuncThatReturnsUint32()
alg := tpm.Algorithm(id)
fmt.Println(alg) // This panics if id was not one of the consts
```

While that is clearly a bug in their code, there's no reason it should panic.

The solution you have here is clever, but clever is bad and risky.